5-Cookies/client-react/src/components/TodoList/TodoListItem/TodoListItem.tsx

```diff
@@ -0,0 +1,83 @@
+import React, { useState, FC } from "react";
+import { ITodoItem, TodoListItem } from "../../../interfaces/interfaces";
+import { classes } from "../../../js-styles/style";
+import IconButton from "../../general/IconButton";
+import Switch from "../../general/Switch";
+import EditTodoItem from "../EditTodoItem";
+import { todosApi } from "../../../../App";
+import dataHooks from "../../../dataHooks/dataHooks";
+import "../../../../style.css";
+import { Item } from "../../../interfaces/interfaces";
+import { Guid } from "../../../../../common/interfaces/Todo";
+
+const TodoListItem: FC<ITodoItem> = ({ item, removeTodo }) => {
+  const [isEditMode, setIsEditMode] = useState<boolean>(false);
+  const [todoItem, setTodoItem] = useState<Item>(item);
+
+  const dispatchEditTodo = async (newTodo: Item, id: Guid): Promise<void> => {
+    try {
+      await todosApi.editTodo(newTodo, id);
+    } catch (error) {
+      console.log(error);
+    }
+  };
+
+  const toggleTodo = async (): Promise<void> => {
+    setTodoItem({ ...todoItem, isFinished: !todoItem.isFinished });
+    dispatchEditTodo({ ...item, isFinished: !todoItem.isFinished }, item.id);
```

Comment:
1. you're not waiting for the async update in `dispatchEditTodo`
2. it might make more sense to update the server first (it is way more error prone), and update the UI afterwards if server has updated successfully. this way you won't get into broken state in the client

Reply:
Alright, I understand your intent, I will follow it